servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpRequestMetaData.java

```diff
@@ -140,7 +142,14 @@ public HttpRequestMetaData requestTarget(final String requestTarget, final Chars
     @Nullable
     @Override
     public final String scheme() {
-        return lazyParseRequestTarget().scheme();
+        if (scheme == null) {
+            final String rawScheme = lazyParseRequestTarget().scheme();
+            if (rawScheme == null) {
+                return null;
+            }
+            scheme = HttpScheme.of(rawScheme);
```

Comment:
The biggest risk is here, if another scheme is used, this method will start throwing `IllegalArgumentException`